test/libcommonserver/CMakeLists.txt

```diff
@@ -9,6 +9,7 @@ set(testcommon_NAME ${APPLICATION_NAME}_test_common)
 set(testcommon_SRCS
     ../test.cpp
     test.cpp
+    ../test_utility/temporarydirectory.h ../test_utility/temporarydirectory.cpp
```

Comment:
Avoid including files outside the current directory (..).
Either put `test_utility`inside `libcommonserver` or create a dedicated CMakeList.txt for this part